src/core/core.datasetController.js

```diff
@@ -287,6 +374,99 @@ helpers.extend(DatasetController.prototype, {
 		});
 	},
 
+	/**
+	 * @private
+	 */
+	_storeParsedData: storeParsedData,
+
+	/**
+	 * @private
+	 */
+	_parse: function(start, count) {
+		var me = this;
+		var meta = me.getMeta();
+		var data = me._data;
+		var i, ilen, item;
+
+		if (!data || data.length === 0) {
+			return;
+		}
+
+		for (i = start, ilen = start + count; i < ilen; ++i) {
```

Comment:
Can we change this so that the for loop is inside the parse methods? Right now it's making a method call for each data point, which is fairly expensive. But if we just check the data type of the first row and assume the rest are the same then we can do the loop in the parse methods, which will be much faster

Reply:
changed